streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -611,9 +616,11 @@ boolean runLoop() {
                     cacheResizer.accept(size);
                 }
                 runOnce();
-                if (nextProbingRebalanceMs.get() < time.milliseconds()) {
+
+                // Check for a scheduled rebalance but don't trigger it until the current rebalance is done
+                if (!taskManager.rebalanceInProgress() && nextProbingRebalanceMs.get() < time.milliseconds()) {
```

Comment:
It seems like this alone is the fix we needed, right?

Maybe a nitpick, but I'm wondering if we really needed the other refactoring/renaming to be part of the same PR.

Reply:
Yes, but in those cases we should actually not have triggered the rebalance anyway -- this should always reflect the latest status of whether a rebalance is actually needed.

And yeah, I'll remove all the thread id stuff but since  there's not much other junk left once that's removed, I would rather keep the minor renaming of `isRebalanceInProgress` if that's ok